components/support/sql/src/conn_ext.rs

```diff
@@ -110,6 +110,10 @@ pub trait ConnExt {
     fn unchecked_transaction(&self) -> SqlResult<UncheckedTransaction> {
         UncheckedTransaction::new(self.conn(), TransactionBehavior::Deferred)
     }
+
+    fn perf_transaction(&self, commit_after: Duration) -> SqlResult<TransactionForPerf> {
```

Comment:
You were right, I'm not the biggest fan of the name. 😂 Maybe `TimeChunkedTransaction`, `TimeSliceTransaction`, `RolledUpTransaction`, or something that makes it clear we're possibly rolling up multiple transactions into one based on time. _Wanders off to paint her bikeshed_ 🚲🎨🏠